File: IOS/DTOs/Objects/Customer/Salon.swift
import Foundation

public enum Salon {
    public enum Responses {}
}

public extension Salon.Responses {
    
    struct Full: Codable, Sendable {
        public var id: UUID
        public var name: String
        public var type: SalonType
        public var description: String?
        public var logo: URL
        public var isActive: Bool
        public var isFavorite: Bool = false
        public var localeId: String
        public var timeZoneId: String
        public var address: Address
        public var point: CoordinatePoint
        
        public init(
            id: UUID,
            name: String,
            type: SalonType,
            description: String? = nil,
            logo: URL,
            isActive: Bool,
            isFavorite: Bool = false,
            localeId: String,
            timeZoneId: String,
            address: Address,
            point: CoordinatePoint
        ) {
            self.id = id
            self.name = name
            self.type = type
            self.description = description
            self.logo = logo
            self.isActive = isActive
            self.isFavorite = isFavorite
            self.localeId = localeId
            self.timeZoneId = timeZoneId
            self.address = address
            self.point = point
        }
    }
}

File: android-legacy/shared/src/main/kotlin/shared/dto/objects/Salon.kt
@file:UseSerializers(UUIDSerializer::class, URISerializer::class)

package shared.dto.objects

import kotlinx.serialization.Serializable
import kotlinx.serialization.UseSerializers
import shared.dto.enums.SalonType
import shared.dto.primitives.Address
import shared.dto.primitives.CoordinatePoint
import shared.dto.protocols.Responsable
import shared.serializers.URISerializer
import shared.serializers.UUIDSerializer
import java.net.URI
import java.util.*

object Salon {
    data object Responses {
        @Serializable
        data class Full(
            var id: UUID,
            var name: String,
            var type: SalonType,
            var description: String? = null,
            var logo: URI? = null,
            var isActive: Boolean,
            var isFavorite: Boolean = false,
            var localeId: String,
            var timeZoneId: String,
            var address: Address,
            var point: CoordinatePoint,
        ) : Responsable
    }
}